lib/jekyll/document.rb

```diff
@@ -180,15 +180,24 @@ def coffeescript_file?
     #   or if the document doesn't contain any Liquid Tags or Variables,
     #   true otherwise.
     def render_with_liquid?
-      !(coffeescript_file? || yaml_file? || json_file? || csv_file?)
+      return false if data["render_with_liquid"] == false
+      return false if coffeescript_file? || yaml_file? || json_file? || csv_file?
+      Utils.has_liquid_construct?(content)
```

Comment:
I broke these two lines up because Rubocop was complaining about line length. This reads better to me anyway.